shared/src/types/vote_extensions/validator_set_update.rs

```diff
@@ -171,10 +189,11 @@ impl VotingPowersMapExt for VotingPowersMap {
             .map(|&(_, &voting_power)| u64::from(voting_power))
             .sum();
 
-        // split the vec into two
-        sorted
-            .into_iter()
-            .map(|(&EthAddress(addr), &voting_power)| {
+        // split the vec into three portions
+        let init = (Vec::new(), Vec::new(), Vec::new());
```

Comment:
nit: think it would be clearer in the code here to unpack into separate variables
```suggestion
        let (hot, cold, voting_powers) = (Vec::new(), Vec::new(), Vec::new());
```